iModelCore/ecobjects/test/NonPublished/CustomAttributeConversionsTest.cpp

```diff
@@ -2133,6 +2133,75 @@ TEST_F(StandardValueToEnumConversionTest, TestRootClassInSeparateSchema)
+            <ECSchemaReference name='baseTest' version='01.00' prefix='base'/>
+            <ECClass typeName='TestClass' isDomainClass='True'>
+                <BaseClass>base:TestClass</BaseClass>
+                <ECProperty propertyName='testProp' typeName='int'/>
+            </ECClass>
+        </ECSchema>)xml";
+
+    Utf8CP schemaXml = R"xml(<?xml version='1.0' encoding='UTF-8'?>
+        <ECSchema schemaName='test' version='1.0' nameSpacePrefix='ts' xmlns='http://www.bentley.com/schemas/Bentley.ECXML.2.0'>
+            <ECSchemaReference name='EditorCustomAttributes' version='01.00' prefix='beca' />
+            <ECSchemaReference name='Middle' version='01.00' prefix='mid'/>
+            <ECClass typeName='TestClass' isDomainClass='True'>
+                <BaseClass>mid:TestClass</BaseClass>
+                <ECProperty propertyName='testProp' typeName='int'>
+                    <ECCustomAttributes>
+                        <StandardValues xmlns='EditorCustomAttributes.01.00'>
+                            <ValueMap>
+                                <ValueMap>
+                                    <Value>0</Value>
+                                    <DisplayString>value0</DisplayString>
+                                </ValueMap>
+                                <ValueMap>
+                                    <Value>1</Value>
+                                    <DisplayString>value1</DisplayString>
+                                </ValueMap>
+                                <ValueMap>
+                                    <Value>2</Value>
+                                    <DisplayString>value2</DisplayString>
+                                </ValueMap>
+                            </ValueMap>
+                        </StandardValues>
+                    </ECCustomAttributes>
+                </ECProperty>
+            </ECClass>
+        </ECSchema>)xml";
+ 
+    ECSchemaPtr baseSchema;
+    ECSchemaPtr middleSchema;
+    ECSchemaPtr schema;
+    ECSchemaReadContextPtr context = ECSchemaReadContext::CreateContext();
+    ASSERT_EQ(SchemaReadStatus::Success, ECSchema::ReadFromXmlString(baseSchema, baseSchemaXml, *context));
+    ASSERT_TRUE(baseSchema.IsValid());
+    ASSERT_EQ(SchemaReadStatus::Success, ECSchema::ReadFromXmlString(middleSchema, middleSchemaXml, *context));
+    ASSERT_TRUE(middleSchema.IsValid());
+    ASSERT_EQ(SchemaReadStatus::Success, ECSchema::ReadFromXmlString(schema, schemaXml, *context));
+    ASSERT_TRUE(schema.IsValid());
+ 
+    EXPECT_TRUE(ECSchemaConverter::Convert(*schema.get(), *context.get())) << "Schema conversion failed";
+ 
+    ASSERT_EQ(0, schema->GetEnumerationCount()) << "The number of enumerations created is not as expected.";
+    ASSERT_EQ(1, baseSchema->GetEnumerationCount()) << "The number of enumerations created is not as expected.";
+
+    CheckTypeName("ts:TestClass_testProp", *schema, "testProp", {"TestClass"});

```

Comment:
We should CheckTypeName on the middleSchema as well, to make sure it's properly wired up. In this case both, the schema and middleSchema use the same alias for "base", you could write another test where they use different aliases and make sure the GetTypeName() method returns the correct one for each level.

We could also add another test where the middle schema does not have its own declaration of testProp, and check if it still behaves the same.

Reply:
Added additional tests